core/src/crypto/SECP256k1Point.cpp

```diff
@@ -36,76 +36,37 @@
 
 namespace ledger {
     namespace core {
+        SECP256k1Point::ContextUniqPtr SECP256k1Point::_context = ContextUniqPtr(secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY), secp256k1_context_destroy);
```

Comment:
Hm, I’m not sure about having that here. Have you considered implications with thread-safe code?

Reply:
Initialization is safe, as this is `static` variable. From docs
```/** Opaque data structure that holds context information (precomputed tables etc.).
 *
 *  The purpose of context structures is to cache large precomputed data tables
 *  that are expensive to construct, and also to maintain the randomization data
 *  for blinding.
 *
 *  Do not create a new context object for each operation, as construction is
 *  far slower than all other API calls (~100 times slower than an ECDSA
 *  verification).
 *
 *  A constructed context can safely be used from multiple threads
 *  simultaneously, but API call that take a non-const pointer to a context
 *  need exclusive access to it. In particular this is the case for
 *  secp256k1_context_destroy and secp256k1_context_randomize.
 *
 *  Regarding randomization, either do it once at creation time (in which case
 *  you do not need any locking for the other calls), or use a read-write lock.
 */
typedef struct secp256k1_context_struct secp256k1_context;```